cmd/utils/flags.go

```diff
@@ -1957,14 +1958,17 @@ func MakeGenesis(ctx *cli.Context) *core.Genesis {
 
 // MakeChain creates a chain manager from set command line flags.
 func MakeChain(ctx *cli.Context, stack *node.Node) (chain *core.BlockChain, chainDb ethdb.Database) {
+	return MakeChainWithEngine(ctx, stack, mockEngine.NewFaker())
+}
+
+func MakeChainWithEngine(ctx *cli.Context, stack *node.Node, engine consensus.Engine) (chain *core.BlockChain, chainDb ethdb.Database) {
 	var err error
 	chainDb = MakeChainDatabase(ctx, stack, false) // TODO(rjl493456442) support read-only database
 	config, _, err := core.SetupGenesisBlock(chainDb, MakeGenesis(ctx))
 	if err != nil {
 		Fatalf("%v", err)
 	}
 	config.FullHeaderChainAvailable = ctx.GlobalString(SyncModeFlag.Name) != "lightest"
-	engine := mockEngine.NewFaker()
```

Comment:
the real issue here is that using a mock consensus engine won't work for a chain. Is there something we can do here @piersy? Like implementing an offline/read only Istanbul Engine?

Reply:
I'd have to look into this more to understand what the import really requires, but my hunch would be that it won't be easy to implement an offline version due to the beast that is `backend.Backend` its 125 methods makes re-implemntation/mocking an overly onerous task.

Long term if we continue to break down and decouple the istanbul code this task could become significantly simpler.

Reply:
I think that if we sort out the config mess, starting the real engine without starting the node is possible. It's actually possible right now, it's just that it will be missing config params that are set in different parts of the code.